bootable-jar/WFLY-14411_Bootable_Jar_Upgrade.adoc

```diff
@@ -0,0 +1,215 @@
+* Galleon feature-packs that are dependencies of the Galleon feature-pack(s) in use to build a Bootable JAR.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14411[WFLY-14411]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-194
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:{jkasik@redhat.com}[Jan Kasik]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [X] QE
+
+=== Affected Projects or Components
+
+* Bootable JAR Maven plugin
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [ ] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+* The upgrade mechanism is done from the Bootable JAR Maven plugin.
+
+* Components that we want to upgrade are referenced with Maven Coordinates 
+(GroupId, ArtifactId, Classifier, Version and Type) from the Maven dependencies section of the Maven project.
+That is the nominal Maven way of expressing dependencies.
+
+* We are not automatically using all artifacts present in the project dependencies section for the upgrade.  
+A user needs to list the GroupId/ArtifactId/[Classifier] of the artifacts he wants to see used during the upgrade.
+
+* The `wildfly-jar-maven-plugin` configuration is evolved with an `<overridden-server-artifacts>` set that contains the artifacts that 
+are in use for the upgrade.
+
+* JBoss Modules jar artifacts and Galleon feature-packs zip artifacts are treated the same way and can be present in the `overridden-server-artifacts` set.
+
+=== Hard Requirements
+
+* This RFE doesn't introduce backward incompatibility.
+
+* This RFE is made possible starting with WildFly 23. Older WildFly release can't be used.
+
+* Usage of Galleon feature-pack location (FPL) to reference the Galleon channel of a producer registered inside a universe) is still possible.
+
+* This mechanism strongly depends on Maven dependency resolution. 
+Any artifact to upgrade must be present in accessible Maven repositories (remote or local cache).
+
+* An artifact present in the  `overridden-server-artifacts` set must be present in the dependencies section. 
+It is advised that the dependency scope be `provided` to be not included in the application artifact (eg: war).  A warning is displayed if
+the scope is not `provided`.
+If an artifact is not present in the dependencies, then a failure occurs during build.
+
+* An artifact is referenced in the `overridden-server-artifacts` by GroupId, ArtifactId and optionally Classifier. 
```

Comment:
@jfdenise What should take precedence if `version` is specified in both `overridden-server-artifats` and `dependencies`? Or should the build fail with an error if this situation occurs?

Reply:
The version in overridden-server-artifacts should take precedence. It allows to have more flexibility.